### PR TITLE
feat(playwright-core): add remove cookies api

### PR DESCRIPTION
**Pull request by PaulTriandafilov**
Implements the changes suggested in https://github.com/microsoft/playwright/issues/29662

This patch contains removeCookies method to remove cookies from BrowserContext by it's names

### CODE REVIEW

File: packages/playwright-core/src/server/browserContext.ts [hunk 1]
```diff
@@ -276,6 +276,15 @@ export abstract class BrowserContext extends SdkObject {
     return await this.doGetCookies(urls as string[]);
   }
 
+  async removeCookies(cookieNames: string | string[]): Promise<void> {
+    const setCookies = await this.cookies();
+
+    const newCookiesSet = setCookies.filter(c => !cookieNames.includes(c.name));
```
Comment (pavelfeldman) [inline packages/playwright-core/src/server/browserContext.ts:282]: It seems like `context.removeCookies('long-cookie-name')` will remove cookies with names `long`, `cookie` and `name`.

File: packages/playwright-core/src/client/browserContext.ts [hunk 2]
```diff
@@ -269,6 +269,13 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
     await this._channel.clearCookies();
   }
 
+  async removeCookies(cookieNames: string | string[]): Promise<void> {
```
Comment (pavelfeldman) [inline packages/playwright-core/src/client/browserContext.ts:272]: The functionality you are implementing is more of a "remove cookies by name". I wonder if you'd like to be a bit more proactive and include optional `domain` and `path` parameters. 

```js
async removeCookies(cookies: { name?: string, domain?: string, path?: string });
```

that way you could be able to clear all cookies by name, all cookies for a given domain, all cookies for a given path and their combinations:

```js
await context.removeCookies({ name: 'session-id' });
await context.removeCookies({ domain: 'my-origin.com' });
await context.removeCookies({ path: '/api/v1' });
await context.removeCookies({ name: 'session-id', domain: 'my-origin.com' });
```

**Review by pavelfeldman — COMMENTED** [review]
Looks great!

File: packages/playwright-core/src/client/browserContext.ts (hunk 2 above)
Comment (PaulTriandafilov) [reply packages/playwright-core/src/client/browserContext.ts]: Thank you for comments! Fixed. Please let me know if there is something else I could change or fix.

File: packages/playwright-core/src/server/browserContext.ts [hunk 3]
```diff
@@ -276,6 +276,22 @@ export abstract class BrowserContext extends SdkObject {
     return await this.doGetCookies(urls as string[]);
   }
 
+  async removeCookies(cookies: {name?: string, domain?: string, path?: string}): Promise<void> {
+    const setCookies = await this.cookies();
+
+    if (!cookies.name && !cookies.domain && !cookies.path)
```
Comment (Smrtnyk) [inline packages/playwright-core/src/server/browserContext.ts:282]: the call above is a waste if you will early return here
maybe moving the `await  this.cookies()` under?

File: packages/playwright-core/src/server/browserContext.ts [hunk 4]
```diff
@@ -276,6 +276,22 @@ export abstract class BrowserContext extends SdkObject {
     return await this.doGetCookies(urls as string[]);
   }
 
+  async removeCookies(cookies: {name?: string, domain?: string, path?: string}): Promise<void> {
```
Comment (Smrtnyk) [inline packages/playwright-core/src/server/browserContext.ts:279]: somehow I feel like `cookies` as name of the argument is misleading
I would expect array of cookies or something
maybe naming it `criteria` or something similar?

File: packages/playwright-core/src/server/browserContext.ts [hunk 5]
```diff
@@ -276,6 +276,22 @@ export abstract class BrowserContext extends SdkObject {
...
+    const setCookies = await this.cookies();
+
+    if (!cookies.name && !cookies.domain && !cookies.path)
+      return;
+
+    const filteredCookies = setCookies.filter(cookie => {
```
Comment (Smrtnyk) [inline packages/playwright-core/src/server/browserContext.ts:285]: maybe naming this `cookiesToKeep` ?

File: packages/playwright-core/src/server/browserContext.ts (hunk 4 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/browserContext.ts:279]: I second this, `filter` would probably be a better name, but we can fix it during the pre-release api review.

File: packages/playwright-core/src/server/browserContext.ts [hunk 6]
```diff
@@ -276,6 +276,22 @@ export abstract class BrowserContext extends SdkObject {
     return await this.doGetCookies(urls as string[]);
   }
 
+  async removeCookies(cookies: {name?: string, domain?: string, path?: string}): Promise<void> {
+    const setCookies = await this.cookies();
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/playwright-core/src/server/browserContext.ts:280]: @Smrtnyk: We discourage stylistic review comments, especially from non-contributors.

File: packages/playwright-core/src/server/browserContext.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/browserContext.ts:282]: This should probably throw 'Either name, domain or path are required'. Same should be mentioned in the doc (md file).

File: packages/playwright-core/src/server/browserContext.ts (hunk 5 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/browserContext.ts:285]: @Smrtnyk: We discourage stylistic review comments, especially from non-contributors.

**Review by pavelfeldman — COMMENTED** [review]
Looks great, a couple of nits and it is ready to go!

File: packages/playwright-core/src/server/browserContext.ts (hunk 5 above)
Comment (PaulTriandafilov) [reply packages/playwright-core/src/server/browserContext.ts]: Got it! just pushed outdated fixes :(
let me quickly get things straightened out

**Comment by PaulTriandafilov** [comment]
@pavelfeldman I guess it's done. Please, let me know if I missed something

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**